Research/ResearchUI/iOS/Views/RSDVerticalGridCollectionView.swift

```diff
@@ -0,0 +1,191 @@
+        }
+    }
+
+    public required init?(coder: NSCoder) {
+        super.init(coder: coder)
+        self.initDefaultFlowLayout()
+    }
+    
+    fileprivate func initDefaultFlowLayout() {
+        let layout = RSDVerticalGridCollectionViewFlowLayout()
+        self.collectionViewLayout = layout
+    }
+}
+
+/// The 'RSDVerticalGridCollectionViewFlowLayout' creates a centered grid layout in conjunction with
+/// the 'RSDVerticalGridCollectionView'.  As the class name implies, this class only works currently in a vertical orientation.
+open class RSDVerticalGridCollectionViewFlowLayout: UICollectionViewFlowLayout {
+            
+    /// The number of columns in each row of the grid
+    open var columnCount: Int = 3
+    
+    /// The width of a cell in the row when it is full at the column count
+    fileprivate var cellWidth: CGFloat {
+        let collectionViewWidth = (collectionView?.bounds.width ?? 0)
+        if collectionViewWidth == 0 {
+            return 0
+        }
+        let width = (collectionViewWidth - (CGFloat(columnCount) * horizontalCellSpacing)) / CGFloat(columnCount)
+        // Round down the value so that we make sure the cells fit in the full width
+        return CGFloat(Int(width))
+    }
+        
+    /// The height of each individual cell
+    open var cellHeightAbsolute: CGFloat = 120
+    /// If non-negative, this will be used  to determine height instead of the absolute height
+    /// Ratio is width * ratio = height
+    open var cellHeightRatio: CGFloat = -1
+    
+    /// If cellHeightRatio is non-negative, it will be used  to determine height instead of the absolute height
+    /// Otherwise the height will be the absolute cellHeight var
+    fileprivate var cellHeight: CGFloat {
+        if cellHeightRatio > 0 {
+            return cellWidth * cellHeightRatio
+        }
+        return cellHeightAbsolute
+    }
+    
+    /// The horizontal spacing between cells
+    open var horizontalCellSpacing: CGFloat = 0 {
+        didSet {
+            self.refreshCellSpacing()
+        }
+    }
+    
+    /// The veritcal spacing between each individual cell
+    open var verticalCellSpacing: CGFloat = 0 {
+        didSet {
+            self.refreshCellSpacing()
+        }
+    }
+    
+    /// The ratio of cellSpacing to cell horizontal border
+    open var cellHorizontalBorderRatio = 0.5
+    
+    /// The number of items in the grid
+    open var itemCount: Int = 0 {
+        didSet {
+            if itemCount > 0 {
+                self.itemSize = self.cellSize(for: IndexPath(row: 0, section: 0))
+            }
+        }
+    }
+    
+    /// This needs to be used for the collection view delegate section count
+    public var sectionCount: Int {
+        return ((itemCount - 1) / columnCount) + 1
+    }
+    
+    /// This needs to be used for the collection view delegate section count
+    public func itemCountInSection(section: Int) -> Int {
+        let lastRowIndex = itemCount / columnCount
```

Comment:
Stick to one or the other. Either:

lastRowIndex -> lastSection

OR

section -> gridrow

Reply:
Went with "gridRow"